### ISSUE TITLE
Incorrect dependency for MailKit/Mimekit 2.10.0

### ISSUE THREAD

**Issue by johnjaylward** [jstedfast/MimeKit#629]
**Describe the bug**
Version 2.10.0 is improperly bringing in System.Data.DataSetExtensions when the target framework is netstandard2.1

**Platform (please complete the following information):**
 - OS: Windows
 - .NET Runtime: multi-tartget library (net461 netstandard2.0 netstandard2.1) netstandard2.1 is the broken one
 - .NET Framework: multi-tartget library (net461 netstandard2.0 netstandard2.1) netstandard2.1 is the broken one
 - MailKit Version: 2.10.0

**To Reproduce**
1. Have a project which implements a DataSet (from System.Data.DataSetExtensions)
2. properly set up the references:

```xml
<!-- not needed for net461 or netstandard2.1 as it's included in those frameworks -->
  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
    <PackageReference Include="System.Data.DataSetExtensions" Version="4.5.0" />
  </ItemGroup>
<!-- for net frameworks, use the framework version -->
  <ItemGroup Condition=" '$(TargetFramework)' == 'net461' ">
    <Reference Include="System.Data.DataSetExtensions" />
  </ItemGroup>
```

3. Include MailKit 2.10.0
4. Rebuild the project. You will see CS0029, CS0030, CS0115, **CS0433**, CS1061, CS1503, and CS1729 errors in the error log.

CS0433 is the most relevant: 
> CS0433	The type 'TypedTableBase<T>' exists in **both 'System.Data.DataSetExtensions, Version=4.0.0.0**, Culture=neutral, PublicKeyToken=b77a5c561934e089' **and 'netstandard, Version=2.1.0.0**, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'

**Expected behavior**
Mailkit should not pull in System.Data.DataSetExtensions for netstandard2.1 as a dependency

**Screenshots**
![image](https://user-images.githubusercontent.com/383703/101057080-a4e89700-3559-11eb-8cac-c779d7601241.png)

**Additional context**
The problem does not exist with MailKit 2.9

**Comment by jstedfast** [comment]
Thanks!

**Comment by johnjaylward** [comment]
Thanks for the quick fix. I was wondering if you'd need to add a specific TargetFramework moniker for netstandard2.1. Is there a public beta build I could try referencing in my project to confirm these changes work?

**Comment by johnjaylward** [comment]
Ok, I found the beta builds on MyGet.

Here's the result of me using it:
![image](https://user-images.githubusercontent.com/383703/101063427-c600b600-3560-11eb-91f6-ea5c76b6ee51.png)

Looks like it's still pulling in the DataSetExtensions package.

Probably need to update the `TargetFrameworks` to include netstandard2.1 explicitly.

**Comment by jstedfast** [comment]
I was wondering about that too. Thanks for confirming.

**Comment by jstedfast** [comment]
This should be fixed now - I just pushed a commit to add netstandard2.1 as a build target.

Let me know if the next myget package does not work for you.

**Comment by jstedfast** [comment]
Looks like the automated build failed due to an intermittent DSA unit test failure. Just kicked off a new build and it should be publishing 2.10.2.7 (or use 2.10.0.8 - looks like my azure-pipelines.yml was using wrong version numbers)

**Comment by jstedfast** [comment]
Ended up releasing 2.10.1.0 to nuget.org with this fix.

**Comment by johnjaylward** [comment]
looks good to me
![image](https://user-images.githubusercontent.com/383703/101393955-da60ed80-3895-11eb-9345-321cfe6efb68.png)

**Comment by johnjaylward** [comment]
can you release an update MailKit as well so it's fixed? I don't think Mailkit itself needs a netstandard2.1 definition, just needs the updated MimeKit

**Comment by jstedfast** [comment]
I'll try to make a release soon - I'm on call this week, so my free time will be extremely limited.

**Comment by johnjaylward** [comment]
> 
> 
> I'll try to make a release soon - I'm on call this week, so my free time will be extremely limited.

No worries. Thanks for the quick fixes. Totally understand the limited time :D

**Comment by jstedfast** [comment]
This is still on my radar, just had the best luck of any guy on the planet. Met this amazing woman while she was standing under a giant Christmas tree and we hit it off and have been spending a ton of time together and I just haven't had much time to work on MimeKit/MailKit this past week. This month has been a crazy whirlwind so far 😃

**Comment by jstedfast** [comment]
Just released 2.10.1 of MailKit
